overlord/restart/restart.go

```diff
@@ -402,6 +457,33 @@ func restartParametersFromChange(chg *state.Change) (*RestartParameters, error)
 	return &rp, nil
 }
 
+func boundaryDirectionFromStatus(status state.Status) RestartBoundaryDirection {
+	switch status {
+	case state.DoStatus, state.DoingStatus, state.DoneStatus:
+		return RestartBoundaryDirectionDo
+	case state.UndoStatus, state.UndoingStatus, state.UndoneStatus:
+		return RestartBoundaryDirectionUndo
+	default:
+		return 0
+	}
+}
+
+func taskIsRestartBoundary(t *state.Task, dir RestartBoundaryDirection) bool {
+	var boundary RestartBoundaryDirection
+	if err := t.Get("restart-boundary", &boundary); err != nil {
+		return false
+	}
+	return (boundary & dir) != 0
+}
+
+// MarkTaskAsRestartBoundary sets a task as a restart boundary. That means
+// the change cannot continue beyond this task, before a restart has taken
+// place. 'path' indicates which execution direction(s) it will be a restart
+// boundary for.
+func MarkTaskAsRestartBoundary(t *state.Task, dir RestartBoundaryDirection) {
```

Comment:
Maybe we need to add a test here to check that it sets the config properly, no? I am guessing it is already being tested implicitly.

Reply:
Good point, might be better to have an explicit test for this one